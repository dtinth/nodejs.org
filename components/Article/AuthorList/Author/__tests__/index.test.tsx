import React from 'react';
import { render } from '@testing-library/react';
import { AppProps } from '../../../../../types';
import { LocaleProvider } from '../../../../../providers/localeProvider';
import messages from '../../../../../i18n/locales/en.json';
import Author from '..';

const i18nData = {
  currentLocale: { code: 'en' },
  localeMessages: messages,
} as unknown as AppProps['i18nData'];

describe('Author component', () => {
  it('renders correctly', () => {
    const username = 'test-author';
    const { container } = render(
      <LocaleProvider i18nData={i18nData}>
        <Author username={username} size="60" />
      </LocaleProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('does not render without a username', () => {
    const { container } = render(
      <LocaleProvider i18nData={i18nData}>
        <Author username="" size="" />
      </LocaleProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
