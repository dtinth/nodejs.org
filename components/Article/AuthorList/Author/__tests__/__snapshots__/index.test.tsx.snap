// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author component does not render without a username 1`] = `
<div>
  <li>
    <a
      aria-label="components.author.githubLinkLabel"
      class="link"
      href="https://github.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span />
    </a>
  </li>
</div>
`;

exports[`Author component renders correctly 1`] = `
<div>
  <li>
    <a
      aria-label="components.author.githubLinkLabel"
      class="link"
      href="https://github.com/test-author"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span />
    </a>
  </li>
</div>
`;
